pydicom/dataset.py

```diff
@@ -208,6 +208,25 @@ def data_element(self, name):
             return self[tag]
         return None
 
+    def raw_data_element(self, name):
```

Comment:
Yeah this is a bit of a hack, but I couldn't find another way to get the offset of the pixel data element, which I needed to efficiently access individual frames.

Access of pixel data within the file is buried deep in the API, which makes it difficult to efficiently read individual frames. One currently has to first explicitly defer reading of the pixel data element and then subsequently re-open the file in `get_pixeldata()` and seek to the start of the pixel data element. This approach is complicated by circular import issues, which are probably caused by the way pixel handler functions get imported and registered. 

Could we refactor the code such that reading the pixel data element value will be deferred by default and the offset to the start of the pixel data element will be stored on the `Dataset` object? This would allow us to load pixel data lazily from the file in `get_pixeldata()` when either the `PixelData` or `pixel_array` attributes are accessed or when the `get_frames()` method is called.

Alternatively, one could read frames directly using lower-level pydicom API functions independently of the `Dataset` class (here exemplified using the *Pillow* library to read pixels):

```python
import pydicom
from PIL import Image
from io import BytesIO


def read_frame_of_encapsulated_pixel_data_element(filename, frame_number):
    '''Reads a single frame from a multi-frame Image Pixel Data element.

    Parameters
    ----------
    filename: str
        name of a DICOM file with an encapsulated multi-frame Image
        Pixel Data element
    frame_number: int
        one-based frame number

    Returns
    -------
    PIL.Image
        pixel data

    '''
    with pydicom.filebase.DicomFile(filename, 'br') as fp:

        # This is only used to find the position of the Pixel Data element
        ds = pydicom.filereader.dcmread(fp, defer_size=0, stop_before_pixels=True)
        fp.is_implicit_VR = ds.is_implicit_VR
        fp.is_little_endian = ds.is_little_endian

        data_element_position = fp.tell()

        # First item is the basic offset table.
        data_element_value_offset = pydicom.filereader.\
             data_element_offset_to_value(fp.is_implicit_VR, 'OB')
        bot_position = (data_element_position + data_element_value_offset)
        fp.seek(bot_position)
        frame_offset_list = pydicom.encaps.get_frame_offsets(fp)

        first_item_position = fp.tell()

        # Determine position of frame within Pixel Data element value.
        try:
            index = frame_number - 1
            frame_offset = frame_offset_list[index]
            next_frame_offset = frame_offset_list[index + 1]
            # Offset is measured from the first byte of the first Item tag
            # following the Basic Offset Table item
            frame_position = first_item_position + frame_offset
        except IndexError:
            error_msg = ("Offset for frame {} could not be determined from "
                         "Basic Offset Table item value.".format(frame_number))
            raise ValueError(error_msg)
        fp.seek(frame_position)
        data = pydicom.encaps.read_item(fp)

    if data is None:
        error_msg = ("Could not read value of Frame item {} "
                     "of Pixel Data element.".format(frame_number))
        raise ValueError(error_msg)

    try:
        image = Image.open(BytesIO(data))
    except:
        error_msg = ("Could not decompress value of Frame item {} "
                     "retrieved from Pixel Data element".format(frame_number))
        raise IOError(error_msg)

    return image
```